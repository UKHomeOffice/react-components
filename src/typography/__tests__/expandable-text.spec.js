import React from 'react'
import { render } from 'enzyme'
import ExpandableText from 'govuk-react-components/typography/expandable-text';

describe('The Expandable Text Component', () => {

  it('should render correctly', () => {
    expect(
      render(<ExpandableText title="Lorem ipsum">Dolor set amet</ExpandableText>)
    ).toMatchSnapshot()
  });

  it('should render correctly when passed a component as a child', () => {
    expect(
      render(<ExpandableText title="Lorem ipsum"><span>Dolor set amet</span></ExpandableText>)
    ).toMatchSnapshot()
  });

  it('should render correctly when passed a component as a title', () => {
    expect(
      render(<ExpandableText title={<span>Lorem ipsum</span>}>Dolor set amet</ExpandableText>)
    ).toMatchSnapshot()
  });

  it('should correctly output the open attr when prop passed', () => {
    expect(
      render(<ExpandableText title="Lorem ipsum" open={true}>Dolor set amet</ExpandableText>)
    ).toMatchSnapshot()
  });

  it('should correctly output the open attr when prop passed', () => {
    expect(
      render(<ExpandableText title="Lorem ipsum" open={true}>Dolor set amet</ExpandableText>)
    ).toMatchSnapshot()
  });

  it('should correctly apply a class when className prop is provided', () => {
    expect(
      render(<ExpandableText title="Lorem ipsum" className="lorem-ipsum">Dolor set amet</ExpandableText>)
    ).toMatchSnapshot()
  });

})