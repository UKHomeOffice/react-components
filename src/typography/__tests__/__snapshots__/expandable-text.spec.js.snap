// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Expandable Text Component should correctly apply a class when className prop is provided 1`] = `
<details
  class="lorem-ipsum"
>
  <summary>
    <span
      class="summary"
    >
      Lorem ipsum
    </span>
  </summary>
  <div
    class="panel panel-border-narrow"
  >
    Dolor set amet
  </div>
</details>
`;

exports[`The Expandable Text Component should correctly output the open attr when prop passed 1`] = `
<details
  open=""
>
  <summary>
    <span
      class="summary"
    >
      Lorem ipsum
    </span>
  </summary>
  <div
    class="panel panel-border-narrow"
  >
    Dolor set amet
  </div>
</details>
`;

exports[`The Expandable Text Component should correctly output the open attr when prop passed 2`] = `
<details
  open=""
>
  <summary>
    <span
      class="summary"
    >
      Lorem ipsum
    </span>
  </summary>
  <div
    class="panel panel-border-narrow"
  >
    Dolor set amet
  </div>
</details>
`;

exports[`The Expandable Text Component should render correctly 1`] = `
<details>
  <summary>
    <span
      class="summary"
    >
      Lorem ipsum
    </span>
  </summary>
  <div
    class="panel panel-border-narrow"
  >
    Dolor set amet
  </div>
</details>
`;

exports[`The Expandable Text Component should render correctly when passed a component as a child 1`] = `
<details>
  <summary>
    <span
      class="summary"
    >
      Lorem ipsum
    </span>
  </summary>
  <div
    class="panel panel-border-narrow"
  >
    <span>
      Dolor set amet
    </span>
  </div>
</details>
`;

exports[`The Expandable Text Component should render correctly when passed a component as a title 1`] = `
<details>
  <summary>
    <span
      class="summary"
    >
      <span>
        Lorem ipsum
      </span>
    </span>
  </summary>
  <div
    class="panel panel-border-narrow"
  >
    Dolor set amet
  </div>
</details>
`;
